### ISSUE TITLE
Minify HTML before caching

### ISSUE THREAD

**Issue by kunstlabdev** [yogthos/Selmer#221]
Is it possible to hook up a minifier before caching the template? 

I was looking at gluing Selmer and this: https://github.com/yogthos/asset-minifier ([more specifically this](https://github.com/Atsman/clj-html-compressor)) but it seems that I would have to run the minifcation on top of Selmer's already cached content?

The thing I am trying to avoid is having to cache twice, or having to run minifcation on requests themselves. 

Or should I manually minify the template itself after each edit?

(did I say minify enough times?)

**Comment by yogthos** [comment]
There isn't any support for minifying the tempates in Selmer itself, but I would be open to adding the option to provide a custom preprocessor function when the templates are read. At the moment you'd have to manage minification manually.

One pattern I tend to use is to have separate resource paths for dev and prod builds. That way you could keep expanded templates at dev time, and have a build hook that minifies them and writes them to the prod resource path at build time.

**Comment by kunstlabdev** [comment]
Thanks for your reply.

After testing a few options I realized that what might work for my use-case won't work for others who want to minify. 

As I am using very minimal templating tools, minification as a pre-processing step works well. But if Selmer is used to generate a lot of diverse html, then minifcation as a post-processing step would probably be desired.

At the moment I will be manually minifying my templates after each change. Maybe this issue should stay open with a feature tag? Otherwise feel free to close.

**Comment by yogthos** [comment]
I don't think it would be possible to do minification on the memoized structure as it's tokenized at that point, and doing minification after the output HTML string is generated would create a performance hit. So, performance would be a concern there. That said, you could already do that now with the HTML string generated by the `render-file` function.

I'm happy to leave an issue as an enhancement though. I can definitely see value in providing pre/post processing hooks.
